### PR TITLE
Add catch error test case.

### PR DESCRIPTION
**Pull request by mshima**
@SBoudrias I'm trying to understand https://github.com/yeoman/yeoman-test/pull/47.
When running this.env.error, errors are been catch twice. Looks like a bug to me.
Should I do something different?

### CODE REVIEW

**Comment by mshima** [comment]
https://github.com/yeoman/yeoman-test/pull/47 is a generator problem https://github.com/yeoman/generator/pull/1155

**Comment by mshima** [comment]
Removed the original use case because it will fail, can be added again once generator 5.5.0 is released.
I didn't squash the PR to keep the build log.

File: lib/run-context.js [hunk 1]
```diff
@@ -21,7 +21,8 @@ var TestAdapter = require('./adapter').TestAdapter;
  *                                       'gen:test' in all cases
  * @param {Object} [settings]
  * @param {Boolean} [settings.tmpdir=true] - Automatically run this generator in a tmp dir
- * @param {Boolean} [settings.compatibility=false] - Run with yeoman-test 1.x compatibility
+ * @param {Boolean} [settings.catchGeneratorError=false] - In some corn cases a error can be thrown but not
```
Comment (SBoudrias) [inline lib/run-context.js]: typo: `corn` -> `corner`

Comment (mshima) [reply lib/run-context.js]:
```suggestion
 * @param {Boolean} [settings.catchGeneratorError=false] - In some corner cases an error can be thrown but not
```

File: test/helpers.js [hunk 2]
```diff
@@ -181,5 +181,57 @@ describe('yeoman-test', function() {
...
+              throw new Error();
+            }
+          }
+        )
+        .on('error', _ => {
+          done();
+        });
+    });
+
+    // This is a workaround for corn case were an error is not correctly emitted
```
Comment (mshima) [inline test/helpers.js]:
```suggestion
    // This is a workaround for corner case were an error is not correctly emitted
```

**Comment by mshima** [comment]
Hold this PR. I will make this env3 only.

**Comment by mshima** [comment]
Probably we should merge https://github.com/yeoman/yeoman-test/pull/67 instead

**Comment by mshima** [comment]
Closing in favor of https://github.com/yeoman/yeoman-test/pull/68
